autoscheduler/frontend/src/tests/ui/Schedule.test.tsx

```diff
@@ -37,24 +38,70 @@ const testMeeting1 = new Meeting({
   section: testSection,
 });
 
-test('Empty schedule renders properly', () => {
-  // arrange and act
-  const store = createStore(autoSchedulerReducer);
-  const { container } = render(<Provider store={store}><Schedule /></Provider>);
+describe('Schedule UI', () => {
+  describe('renders without errors', () => {
+    test('when given an empty schedule', () => {
+      // arrange and act
+      const store = createStore(autoSchedulerReducer);
+      const { container } = render(<Provider store={store}><Schedule /></Provider>);
 
-  // assert
-  expect(container).toBeTruthy();
-});
+      // assert
+      expect(container).toBeTruthy();
+    });
+
+    test('when given a schedule with 1 meeting', () => {
+      // arrange and act
+      const store = createStore(autoSchedulerReducer, {
+        schedules: [[testMeeting1]], selectedSchedule: 0,
+      });
+      const { container } = render(
+        <Provider store={store}>
+          <Schedule />
+        </Provider>,
+      );
+
+      // assert
+      expect(container).toBeTruthy();
+    });
+  });
+
+  describe('uses unique colors', () => {
+    test('for up to 10 different sections', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, {
+        schedules: [testSchedule3], selectedSchedule: 0,
+      });
+      const { getAllByText } = render(
+        <Provider store={store}>
+          <Schedule />
+        </Provider>,
+      );
+
+      // act
+      const cardEls = getAllByText(/\w{4} \d{3}-\d{3}/);
+      const meetingCards = cardEls.map((card) => {
+        // find the colored element that is a parent of the div with text
+        let coloredEl = card;
+        while (!coloredEl.style.backgroundColor) coloredEl = coloredEl.parentElement;
+
+        return {
+          backgroundColor: coloredEl.style.backgroundColor,
+          textContent: card.textContent,
+        };
+      });
+      const uniqueSections = new Set(testSchedule3.map((mtg) => mtg.section.id));
 
-test('Schedule with one meeting renders properly', () => {
-  // arrange and act
-  const store = createStore(autoSchedulerReducer, { meetings: [testMeeting1] });
-  const { container } = render(
-    <Provider store={store}>
-      <Schedule />
-    </Provider>,
-  );
-
-  // assert
-  expect(container).toBeTruthy();
+      // assert
+      // there are 10 unique colors
+      expect(uniqueSections.size).toBeLessThanOrEqual(10);
```

Comment:
Shouldn't this be `toBeGreaterThanOrEqual(10)`? Otherwise this would pass when there are less than 10 unique colors

Reply:
I'm actually using this to assert that the test data hasn't been changed to add more than 10 sections. Because if that were the case, then I would loop back and start re-using colors.

Reply:
It's more of a pre-condition than a post-condition, so it might even make sense to check this in the arrange phase

Reply:
yeah i figured it was something weird like that since it was checking `uniqueSections`, if's not actually part of the assert then I'd move it up and add a comment explaining it